code/modules/client/preference_setup/occupation/occupation.dm

```diff
@@ -72,95 +107,175 @@
 
 /datum/category_item/player_setup_item/occupation/OnTopic(href, href_list, user)
 	if(href_list["reset_jobs"])
 		ResetJobs()
-		pref.req_update_icon = 1
 		return TOPIC_REFRESH
 
 	else if(href_list["job_alternative"])
 		if(pref.alternate_option == GET_RANDOM_JOB || pref.alternate_option == BE_ASSISTANT)
 			pref.alternate_option += 1
 		else if(pref.alternate_option == RETURN_TO_LOBBY)
 			pref.alternate_option = 0
-		pref.req_update_icon = 1
 		return TOPIC_REFRESH
 
+	else if(href_list["select_alt_title"])
+		var/datum/job/job = locate(href_list["select_alt_title"])
+		if (job)
+			var/choices = list(job.title) + job.alt_titles
+			var/choice = input("Choose an title for [job.title].", "Choose Title", pref.GetPlayerAltTitle(job)) as anything in choices|null
+			if(choice && CanUseTopic(user))
+				SetPlayerAltTitle(job, choice)
+				return (pref.equip_preview_mob ? TOPIC_REFRESH_UPDATE_PREVIEW : TOPIC_REFRESH)
+
 	else if(href_list["set_job"] && href_list["set_level"])
-		SetJob(user, href_list["set_job"], text2num(href_list["set_level"]))
+		if(SetJob(user, href_list["set_job"], text2num(href_list["set_level"]))) return (pref.equip_preview_mob ? TOPIC_REFRESH_UPDATE_PREVIEW : TOPIC_REFRESH)
+/*
+	else if(href_list["set_skills"])
+		var/rank = href_list["set_skills"]
+		var/datum/job/job = SSjob.GetJob(rank)
+		if(job)
+			open_skill_setup(user, job)
+*/
+	//From the skills popup
+	/*
+	else if(href_list["hit_skill_button"])
+		var/decl/hierarchy/skill/S = locate(href_list["hit_skill_button"])
+		var/datum/job/J = locate(href_list["at_job"])
+		if(!istype(J))
+			return
+		var/value = text2num(href_list["newvalue"])
+		update_skill_value(J, S, value)
+		pref.ShowChoices(user) //Manual refresh to allow us to focus the panel, not the main window.
+		panel.set_content(generate_skill_content(J))
+		panel.open()
+		winset(user, panel.window_id, "focus=1") //Focuses the panel.
+
+	else if(href_list["skillinfo"])
+		var/decl/hierarchy/skill/S = locate(href_list["skillinfo"])
+		if(!istype(S))
+			return
+		var/HTML = list()
+		HTML += "<h2>[S.name]</h2>"
+		HTML += "[S.desc]<br>"
+		var/i
+		for(i=SKILL_MIN, i <= SKILL_MAX, i++)
+			var/level_name = S.levels[i]
+			HTML +=	"<br><b>[level_name]</b>: [S.levels[level_name]]<br>"
+		show_browser(user, jointext(HTML, null), "window=\ref[user]skillinfo")
+	*/
+	else if(href_list["job_info"])
+		var/rank = href_list["job_info"]
+		var/datum/job/job = SSjob.GetJob(rank)
+
+		job_desc = ""
+		
+		job_desc += "<p style='margin-top:0px; background-color: [job.selection_color]'><br><p>"
+		if(job.alt_titles)
+			job_desc += "<i><b>Alternative titles:</b> [english_list(job.alt_titles)].</i>"
+		send_rsc(user, job.get_job_icon(), "job[ckey(rank)].png")
+		job_desc += "<img src=job[ckey(rank)].png width=96 height=96 style='float:left;'>"
+		if(job.department)
+			job_desc += "<b>Department:</b> [job.department]."
+			if(job.head_position)
+				job_desc += "You are in charge of this department."
```

Comment:
Space between job name and that line is missed.

Reply:
You mean line break ? im not sure if it needed since job info was packed inside small window that not resizable. We'll leave it to @NanakoAC to decide since she is busy with descriptions.

Reply:
ah i noticed my name, hii

The descriptions i'm writing are quite long, the textbox will need to be scrollable